golang/cosmos/cmd/agd/main.go

```diff
@@ -13,7 +13,7 @@ import (
 
 func main() {
 	// We need to delegate to our default app for running the actual chain.
-	daemoncmd.OnStartHook = func(logger log.Logger) {
+	daemoncmd.OnStartHook = func(logger log.Logger, daemonize bool) {
```

Comment:
I dislike the use of a boolean to distinguish `agd start` from `agd export`, especially since then the names `OnStartHook` and `daemonize` are pretty misleading.  Can you instead just add `daemoncmd.OnExportHook` and leave the meaning of `daemoncmd.OnStartHook` alone?

This `main.go` could use this same closure for both:
```suggestion
	launchVM := func(logger log.Logger) {
		...
	}
	daemoncmd.OnExportHook = launchVM
	daemoncmd.OnStartHook = launchVM
```